checkstyle.xml

```diff
@@ -95,6 +95,8 @@
       <property name="illegalPkgs" value="junit.framework"/>
       <property name="illegalClasses" value="java.io.File,java.io.FileInputStream,java.io.FileOutputStream,java.io.FileReader,java.io.FileWriter,java.io.RandomAccessFile,java.util.concurrent.Semaphore,java.util.concurrent.CountDownLatch,java.util.concurrent.Executors,java.util.concurrent.LinkedBlockingQueue,java.util.concurrent.SynchronousQueue,java.util.concurrent.ArrayBlockingQueue,com.google.common.util.concurrent.Futures,java.util.concurrent.CompletableFuture,io.netty.util.concurrent.Future,io.netty.util.concurrent.Promise,io.netty.util.concurrent.AbstractFuture,com.google.common.util.concurrent.ListenableFutureTask,com.google.common.util.concurrent.ListenableFuture,com.google.common.util.concurrent.AbstractFuture"/>
     </module>
+    <module name="RedundantImport"/>
```

Comment:
@maoling why did you put these modules exactly just here? What's wrong with putting them at the end?

Reply:
- Because I want to put all the modules about import together with `IllegalImport`, Now I have put the new added modules at the end of `checkstyle.xml`
- I take a look at the discussions under the [JIRA](https://issues.apache.org/jira/browse/CASSANDRA-17876), if any other review comments I will process asap.

Reply:
ok to the first bullet point.